### PR TITLE
Allow custom commit message when publishing a GitHub Pull Request

### PR DESCRIPTION
**Pull request by pballandras**
## Hey, I just made a Pull Request! 👋 

Currently, in the `publish:github:pull-request` built-in scaffolder action, you specify a pull request title. This pull request title is then used as the commit message for the commit done in the context of the PR (oh and also for the PR title, who would have guessed 😜). 

In our instance of Backstage, we have a template that would require a commit message that differs from the pull request's title. This is for various commit conventions and/or CI. This is currently not doable and this is what led to this pull request.

This pull request adds a new option in the inputs for the action that allows users to specify a custom commit message. The new behaviour is to check if the commit message option is set and to fallback to the pull request title if it is not. In other words, it's a non breaking change that can be released without affecting anyone currently using the action.

tl;dr: In the `publish:github:pull-request` built-in action for the scaffolder, it adds an option to specify a custom, different commit message instead of it being the PR title. It is non breaking since it will fallback to the title like before when unset.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/githubPullRequest.ts [hunk 1]
```diff
@@ -202,6 +203,11 @@ export const createPublishGithubPullRequestAction = (
             description:
               'The teams that will be added as reviewers to the pull request',
           },
+          commitMessage: {
+            type: 'string',
+            title: 'Commit Message',
+            description: 'The commit message for the PR commit',
```
Comment (dotboris) [inline plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/githubPullRequest.ts:209]: To match the rest of the schema

```suggestion
            description: 'The commit message for the pull request commit',
```

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/githubPullRequest.ts [hunk 2]
```diff
@@ -298,7 +305,7 @@ export const createPublishGithubPullRequestAction = (
           changes: [
             {
               files,
-              commit: title,
+              commit: commitMessage || title,
```
Comment (dotboris) [inline plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/githubPullRequest.ts:308]: I think that you can use the null coalescence operator here.

```suggestion
              commit: commitMessage ?? title,
```

**Review by dotboris — APPROVED** [review]

**Review by Rugvip — APPROVED** [review]
Nice, thank you! 👍 🎉 

Sorry this took a while to review
